tests/test_hresult.py

```diff
@@ -113,3 +113,41 @@ def test_pytket():
     )
 
     assert pytket_result == expected
+
+
+def test_collate_tag():
+    # test use of same tag for all entries of array
+
+    shotlist = []
+    for _ in range(10):
+        shot = HResult()
+        _ = [
+            shot.append(reg, 1)
+            for reg, size in (("c", 3), ("d", 5))
+            for _ in range(size)
+        ]
+        shotlist.append(shot)
+
+    weird_shot = HResult((("c", 1), ("d", 1), ("d", 0), ("e", 1)))
+    assert weird_shot.collate_tags() == {"c": [1], "d": [1, 0], "e": [1]}
+
+    lst_shot = HResult([("lst", [1, 0, 1]), ("lst", [1, 0, 1])])
+    shots = HShots([*shotlist, weird_shot, lst_shot])
+
+    counter = shots.collated_counts()
+
+    assert counter == {
+        "c": Counter({"111": 10, "1": 1}),
+        "d": Counter({"11111": 10, "10": 1}),
+        "e": Counter({"1": 1}),
+        "lst": Counter({"101101": 1}),
+    }
```

Comment:
In this counter, are we losing information about the correlation between different tags across shots?

Reply:
yes, but it is still preserved in the full results object. The conversion to pytket preserves it for example.

Reply:
I see. I think those correlations would be important in most cases. No harm in offering this format, but I think we should also offer a counter-like result format that doesn't lose any information _except_ for the shot ordering. I guess this would have to be something like `Counter({(("c", "111"), ("d", "10")): 1, ...})`?

Reply:
yeah maybe that makes more sense as a default

Reply:
Hmm, although even that format isn't sufficient to capture the order of results within a shot (e.g. whether the "d" came between the first and second "c" or not), representing different control-flow paths. For that, we may need a format like `Counter({("c", 1), ("c", 1), ("d", 1), ("c", 1), ("d", 0)) : 1, ...})`.

Reply:
that's too noisy to be useful in this context I think

Reply:
We may need to offer several different "views" into results to serve different use cases.

Reply:
yep, we already have `register_counts` for serving the traditional "classical register" style use, this RFC is effectively view number 2

Reply:
Sounds good to me.

Reply:
> representing different control-flow paths
I think `Counter(shots.results)` where `shots` is an instance of `HShots` already gets you this